overlord/assertstate/assertmgr.go

```diff
@@ -86,3 +105,182 @@ func Add(s *state.State, a asserts.Assertion) error {
 	// TODO: deal together with asserts itself with (cascading) side effects of possible assertion updates
 	return cachedDB(s).Add(a)
 }
+
+// TODO: snapstate also has this, move to auth, or change a bit the approach now that we have AuthContext in the store?
+func userFromUserID(st *state.State, userID int) (*auth.UserState, error) {
+	if userID == 0 {
+		return nil, nil
+	}
+	return auth.User(st, userID)
+}
+
+// fetch fetches or updates the referenced assertion and all its prerequisites from the store and adds them to the system assertion database. It does not fail if required assertions were already present.
+func fetch(s *state.State, ref *asserts.Ref, userID int) error {
+	// TODO: once we have a bulk assertion retrieval endpoint this approach will change
+
+	user, err := userFromUserID(s, userID)
+	if err != nil {
+		return err
+	}
+
+	db := cachedDB(s)
+	store := snapstate.Store(s)
+
+	s.Unlock()
+	defer s.Lock()
+
+	got := ([]asserts.Assertion)(nil)
```

Comment:
I'm just curious, why not just `got := []asserts.Assertion{}` ? Because this is more efficient (no allocation if we don't get anything)?